src/App.tsx

```diff
@@ -97,18 +99,21 @@ const App = () => {
     };
 
     let result = generateResultString(settings);
-
+    if (autoCopy) {
+        navigator.clipboard.writeText(result);
+    }
     return (
         <div className="wrapper">
             <div className="container">
                 <div className="item-wide info-header">Path of Exile - Vendor search tool</div>
                 <div className="item-wide">
                     <div className="result-box">
-                        <div className={result.length > maxLenght ? "result" : result === copied ? "result copied-good" : "result"}>
+                        <div className={result.length > maxLenght ? "result" : result === copied || autoCopy ? "result copied-good" : "result"}>
```

Comment:
A future improvement could be implementing react lifecycle methods, in which case we could remove this here and use setCopied rather in a lifecycle method